include/cuco/detail/hash_functions/xxhash.cuh

```diff
@@ -236,38 +250,47 @@ struct XXHash_64 {
    */
   constexpr result_type __host__ __device__ operator()(Key const& key) const noexcept
   {
-    // TODO do we need to add checks/hints for alignment?
-    constexpr auto nbytes             = sizeof(Key);
-    [[maybe_unused]] auto const bytes = reinterpret_cast<char const*>(&key);  ///< per-byte access
-    [[maybe_unused]] auto const blocks4 =
-      reinterpret_cast<std::uint32_t const*>(&key);  ///< 4-byte word access
-    [[maybe_unused]] auto const blocks8 =
-      reinterpret_cast<std::uint64_t const*>(&key);  ///< 8-byte word access
+    return (*this)(key, cuco::experimental::extent<std::size_t, sizeof(Key)>{});
+  }
+
+  /**
+   * @brief Returns a hash value for its argument, as a value of type `result_type`.
+   *
+   * @tparam Extent The extent type
+   *
+   * @param key The input argument to hash
+   * @param size The extent of the key in bytes
+   * @return A resulting hash value for `key`
+   */
+  template <typename Extent>
+  constexpr result_type __host__ __device__ operator()(Key const& key, Extent size) const noexcept
```

Comment:
This feels weird. Normally, I'd expect such an esoteric use case to provide a custom hash function as opposed to trying to twist the standard hash function signature to make it work for this kind of situation.

Reply:
Just discussed this with @PointKernel. I will change the signature to 
```cpp
template <typename Extent>
constexpr result_type __host__ __device__ hash_bytes(void* data, Extent size) const noexcept
```

Reply:
The idea is that a user can use this function to build a custom hasher around our hashers.

Reply:
A bit more modern approach would be
```cpp
template<class T, std::size_t N>
constexpr result_type __host__ __device__ compute_hash(cuda::std::span<T, N> data) const noexcept;
```

Reply:
Addressed in [74a3739](https://github.com/NVIDIA/cuCollections/pull/327/commits/74a3739df0719c66e57d4162703d630f3088946e)